pennylane/transforms/decompositions/two_qubit_unitary.py

```diff
@@ -183,19 +184,20 @@ def _su2su2_to_tensor_products(U):
 
     # Next, extract B. Can do from any of the C, just need to be careful in
     # case one of the elements of A is 0.
-    # We use B1 unless division by 0 has caused all elements to be inf.
+    # We use B1 unless division by 0 would cause all elements to be inf.
+    use_B2 = math.allclose(A[0, 0], 0.0, atol=1e-6)
     if not math.is_abstract(A):
-        if not math.allclose(A[0, 0], 0.0, atol=1e-6):
+        if not use_B2:
             B = C1 / math.cast_like(A[0, 0], 1j)
         else:
             B = C2 / math.cast_like(A[0, 1], 1j)
     else:
-        B1 = C1 / math.cast_like(A[0, 0], 1j)
-        B2 = C2 / math.cast_like(A[0, 1], 1j)
-        B1_and_B2 = math.array([B1, B2], like="jax")
-        # jit does not like int(bool), but bool + 0 is a supported way of converting True/False to 1/0
-        idx = math.allclose(A[0, 0], 0.0, atol=1e-6) + 0
-        B = B1_and_B2[idx]
+        B = jax.lax.cond(
```

Comment:
I think here you will need to register the function in the math module, we don't install Jax by default, but the solution looks good.

Reply:
Potentially it could also be a TF object for autograph, so we need to update the condition.

Reply:
Turns out it was already registered, so `qml.math.cond` already works! 

Added an interface check.